### DISCUSSION TITLE
Actors don't wake up when touching actor gets removed

### DISCUSSION THREAD

**Discussion by jankrassnigg** [jrouwe/JoltPhysics#270] [Q&A]
I have a test case where an actor falls onto another one and pushes it down. The other actor has a constraint with a motor that should push it up. After a while the top actor gets removed and thus the bottom one should move back to its desired location, however it doesn't, as long as nothing wakes it up, it just stays in the deformed location.

Is this intentional? If I want this kind of behavior, would I need to manually wake up bodies before removing them (to wake up their island) ?

**Comment by jrouwe** [comment]
This is intentional since waking up bodies can be very costly if you're e.g. removing an entire level section. If you want to wake up surrounding bodies do a broad phase query with the bounding box of the body you're removing and wake up what is returned

**Reply by AndreaCatania** [reply]
Oh, I ran into the same issue, good to know.

**Reply by AndreaCatania** [reply]
Btw, also teleporting a body doesn't cause the other bodies to waken-up, so I'm doing it manually.
